agent-bridge/src/main/java/com/newrelic/agent/bridge/DefaultCollectionFactory.java

```diff
@@ -11,14 +11,33 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
 
+/**
+ * This implementation of {@link CollectionFactory} will only be used if the agent-bridge
+ * is being used by an application and the agent is NOT being loaded. Thus, it is unlikely
+ * that the objects created by this implementation are going to receive much use.
+ * So methods in this implementation do not need to implement all functional requirements
+ * of the methods in the interface, but they should not break under low use.
+ */
 public class DefaultCollectionFactory implements CollectionFactory {
 
     @Override
     public <K, V> Map<K, V> createConcurrentWeakKeyedMap() {
         return Collections.synchronizedMap(new WeakHashMap<K, V>());
     }
 
+    @Override
+    public <K, V> Function<K, V> memoize(Function<K, V> loader, int maxSize) {
+        Map<K, V> map = new ConcurrentHashMap<>();
+        return k -> map.computeIfAbsent(k, k1 -> {
```

Comment:
It's a little more code, but you could do this: 

        return k -> {
            if (map.size() >= maxSize) {
                V value = map.get(k);
                return value == null ? loader.apply(k) : value;
            }
            return map.computeIfAbsent(k, loader::apply);
        };

The performance would be slightly better once you reach the `maxSize` because you wouldn't have to keep creating iterators and removing elements from the map.

Reply:
This class is never used by the agent.

It is just there as a placeholder to prevent blowing up an application that uses the agent bridge without the agent.

But indeed, the suggestion is a good one. I will incorporate that in the parent of this PR.